### DISCUSSION TITLE
Migrate from Redis to ValKey or KeyDB

### DISCUSSION THREAD

**Discussion by caniko** [immich-app/immich#10087] [General]
Redis performed a [rug pull](https://openpath.chadwhitacre.com/2024/relicensing-and-rug-pulls/) on its FOSS community. You should either switch to [ValKey](https://github.com/valkey-io/valkey) if you want feature parity with Redis, or move to [KeyDB](https://github.com/Snapchat/KeyDB) the Snapchat fork that makes Redis multithreaded.

**Comment by bo0tzz** [comment]
We're planning to remove redis in favour of using postgres for the queues, so we won't be putting any effort into migrating to alternatives.

**Reply by OnixGH** [reply]
But according to your cursed knowledge you moved back to redis?
https://github.com/immich-app/immich/pull/10801

**Reply by caniko** [reply]
For a queue, maybe [NATS](https://docs.nats.io/nats-concepts/jetstream) is the way to go?

**Reply by alextran1502** [reply]
Since Redis has always been in the stack, we are reusing the piece until we have a better solution

**Reply by caniko** [reply]
Did you consider NATS though @alextran1502 ?

**Comment by alextran1502** [comment]
No I did not, we want to reduce the deployment setup 😅

**Comment by zatricky** [comment]
Just adding some data here ; For a while I'd been running with eqalpha/keydb as a drop-in replacement with no visible issues. I've switched to valkey/valkey now because I was concerned that that keydb image on dockerhub was not updated in over 10 months. Valkey appears to work fine as a drop-in replacement too.
